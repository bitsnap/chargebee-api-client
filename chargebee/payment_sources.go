package chargebee

// THIS IS GENERATED CODE. DO NOT EDIT.

import (
	"net/url"

	"github.com/bitsnap/chargebee-api-client/chargebee/enums"
	"github.com/bitsnap/chargebee-api-client/chargebee/models"

	"github.com/goccy/go-json"
	"github.com/gofiber/fiber/v2"
)

type PaymentSource struct {
	Id               string                   `json:"id" validate:"required"`
	ResourceVersion  int64                    `json:"resource_version"`
	UpdatedAt        uint64                   `json:"updated_at"`
	CreatedAt        uint64                   `json:"created_at" validate:"required"`
	CustomerId       string                   `json:"customer_id" validate:"required"`
	Type             enums.TypeEnum           `json:"type" validate:"required"`
	ReferenceId      string                   `json:"reference_id" validate:"required"`
	Status           enums.StatusEnum         `json:"status" validate:"required"`
	Gateway          enums.GatewayEnum        `json:"gateway" validate:"required"`
	GatewayAccountId string                   `json:"gateway_account_id"`
	IpAddress        string                   `json:"ip_address"`
	IssuingCountry   string                   `json:"issuing_country"`
	Deleted          bool                     `json:"deleted" validate:"required"`
	BusinessEntityId string                   `json:"business_entity_id"`
	Card             models.Card              `json:"card"`
	BankAccount      models.BankAccount       `json:"bank_account"`
	Boleto           models.CustVoucherSource `json:"boleto"`
	BillingAddress   models.BillingAddress    `json:"billing_address"`
	AmazonPayment    models.AmazonPayment     `json:"amazon_payment"`
	Upi              models.Upi               `json:"upi"`
	Paypal           models.Paypal            `json:"paypal"`
	Venmo            models.Venmo             `json:"venmo"`
	KlarnaPayNow     models.KlarnaPayNow      `json:"klarna_pay_now"`
	Mandates         []models.Mandate         `json:"mandates"`
}

func ListPaymentSourcesPageSortBy(site string, offset string, sortBy *SortBy) ([]PaymentSource, string, error) {
	client := fiber.AcquireClient()
	defer fiber.ReleaseClient(client)

	parsedUrl, err := url.Parse("https://" + site + "." + "chargebee.com/api/v2/payment_sources")
	if err != nil {
		return nil, "", err
	}

	content, err := getQuery(client, parsedUrl, offset, sortBy)
	if err != nil {
		return nil, "", err
	}

	type PaymentSourceListItem struct {
		PaymentSource PaymentSource `json:"PaymentSource"`
	}

	type PaymentSourcePage struct {
		List       []PaymentSourceListItem `json:"list"`
		NextOffset string                  `json:"next_offset,omitempty"`
	}

	entries := PaymentSourcePage{
		List: make([]PaymentSourceListItem, 0, 10),
	}

	err = json.Unmarshal(content, &entries)
	if err != nil {
		return nil, "", err
	}

	if len(entries.List) == 0 {
		return []PaymentSource{}, "", nil
	}

	result := make([]PaymentSource, 0, len(entries.List))
	for _, r := range entries.List {
		result = append(result, r.PaymentSource)
	}

	if len(entries.NextOffset) > 0 {
		return resultWithOffset(result, offset, entries.NextOffset)
	}

	return result, "", nil
}

func ListPaymentSourcesPage(site string, offset string) ([]PaymentSource, string, error) {
	return ListPaymentSourcesPageSortBy(site, offset, nil)
}

func RetrievePaymentSourceById(site string, id string) (*PaymentSource, error) {
	client := fiber.AcquireClient()
	defer fiber.ReleaseClient(client)

	parsedUrl, err := url.Parse("https://" + site + "." + "chargebee.com/api/v2/payment_sources/" + id)
	if err != nil {
		return nil, err
	}

	content, err := getQuery(client, parsedUrl, "", nil)
	if err != nil {
		return nil, err
	}

	type PaymentSourceItem struct {
		PaymentSource PaymentSource `json:"PaymentSource"`
	}

	var item PaymentSourceItem

	err = json.Unmarshal(content, &item)
	if err != nil {
		return nil, err
	}

	return &item.PaymentSource, nil
}
