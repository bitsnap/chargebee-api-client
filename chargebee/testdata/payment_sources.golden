# Generated by goldga. DO NOT EDIT.
[snapshots]
"Chargebee payment sources when accessing payment sources API should be able list payment sources" = '''
([]chargebee.PaymentSource) (len=5) {
 (chargebee.PaymentSource) {
  Id: (string) "",
  ResourceVersion: (int64) 0,
  UpdatedAt: (uint64) 0,
  CreatedAt: (uint64) 0,
  CustomerId: (string) "",
  Type: (enums.TypeEnum) "",
  ReferenceId: (string) "",
  Status: (enums.StatusEnum) "",
  Gateway: (enums.GatewayEnum) "",
  GatewayAccountId: (string) "",
  IpAddress: (string) "",
  IssuingCountry: (string) "",
  Deleted: (bool) false,
  BusinessEntityId: (string) "",
  Card: (models.Card) {
  },
  BankAccount: (models.BankAccount) {
  },
  Boleto: (models.CustVoucherSource) {
  },
  BillingAddress: (models.BillingAddress) {
   Address: (models.Address) {
    FirstName: (string) "",
    LastName: (string) "",
    Email: (string) "",
    Company: (string) "",
    Phone: (string) "",
    Line1: (string) "",
    Line2: (string) "",
    Line3: (string) "",
    City: (string) "",
    StateCode: (string) "",
    State: (string) "",
    Country: (string) "",
    Zip: (string) "",
    ValidationStatus: (enums.ValidationStatusEnum) ""
   }
  },
  AmazonPayment: (models.AmazonPayment) {
  },
  Upi: (models.Upi) {
  },
  Paypal: (models.Paypal) {
  },
  Venmo: (models.Venmo) {
  },
  KlarnaPayNow: (models.KlarnaPayNow) {
  },
  Mandates: ([]models.Mandate) <nil>
 },
 (chargebee.PaymentSource) {
  Id: (string) "",
  ResourceVersion: (int64) 0,
  UpdatedAt: (uint64) 0,
  CreatedAt: (uint64) 0,
  CustomerId: (string) "",
  Type: (enums.TypeEnum) "",
  ReferenceId: (string) "",
  Status: (enums.StatusEnum) "",
  Gateway: (enums.GatewayEnum) "",
  GatewayAccountId: (string) "",
  IpAddress: (string) "",
  IssuingCountry: (string) "",
  Deleted: (bool) false,
  BusinessEntityId: (string) "",
  Card: (models.Card) {
  },
  BankAccount: (models.BankAccount) {
  },
  Boleto: (models.CustVoucherSource) {
  },
  BillingAddress: (models.BillingAddress) {
   Address: (models.Address) {
    FirstName: (string) "",
    LastName: (string) "",
    Email: (string) "",
    Company: (string) "",
    Phone: (string) "",
    Line1: (string) "",
    Line2: (string) "",
    Line3: (string) "",
    City: (string) "",
    StateCode: (string) "",
    State: (string) "",
    Country: (string) "",
    Zip: (string) "",
    ValidationStatus: (enums.ValidationStatusEnum) ""
   }
  },
  AmazonPayment: (models.AmazonPayment) {
  },
  Upi: (models.Upi) {
  },
  Paypal: (models.Paypal) {
  },
  Venmo: (models.Venmo) {
  },
  KlarnaPayNow: (models.KlarnaPayNow) {
  },
  Mandates: ([]models.Mandate) <nil>
 },
 (chargebee.PaymentSource) {
  Id: (string) "",
  ResourceVersion: (int64) 0,
  UpdatedAt: (uint64) 0,
  CreatedAt: (uint64) 0,
  CustomerId: (string) "",
  Type: (enums.TypeEnum) "",
  ReferenceId: (string) "",
  Status: (enums.StatusEnum) "",
  Gateway: (enums.GatewayEnum) "",
  GatewayAccountId: (string) "",
  IpAddress: (string) "",
  IssuingCountry: (string) "",
  Deleted: (bool) false,
  BusinessEntityId: (string) "",
  Card: (models.Card) {
  },
  BankAccount: (models.BankAccount) {
  },
  Boleto: (models.CustVoucherSource) {
  },
  BillingAddress: (models.BillingAddress) {
   Address: (models.Address) {
    FirstName: (string) "",
    LastName: (string) "",
    Email: (string) "",
    Company: (string) "",
    Phone: (string) "",
    Line1: (string) "",
    Line2: (string) "",
    Line3: (string) "",
    City: (string) "",
    StateCode: (string) "",
    State: (string) "",
    Country: (string) "",
    Zip: (string) "",
    ValidationStatus: (enums.ValidationStatusEnum) ""
   }
  },
  AmazonPayment: (models.AmazonPayment) {
  },
  Upi: (models.Upi) {
  },
  Paypal: (models.Paypal) {
  },
  Venmo: (models.Venmo) {
  },
  KlarnaPayNow: (models.KlarnaPayNow) {
  },
  Mandates: ([]models.Mandate) <nil>
 },
 (chargebee.PaymentSource) {
  Id: (string) "",
  ResourceVersion: (int64) 0,
  UpdatedAt: (uint64) 0,
  CreatedAt: (uint64) 0,
  CustomerId: (string) "",
  Type: (enums.TypeEnum) "",
  ReferenceId: (string) "",
  Status: (enums.StatusEnum) "",
  Gateway: (enums.GatewayEnum) "",
  GatewayAccountId: (string) "",
  IpAddress: (string) "",
  IssuingCountry: (string) "",
  Deleted: (bool) false,
  BusinessEntityId: (string) "",
  Card: (models.Card) {
  },
  BankAccount: (models.BankAccount) {
  },
  Boleto: (models.CustVoucherSource) {
  },
  BillingAddress: (models.BillingAddress) {
   Address: (models.Address) {
    FirstName: (string) "",
    LastName: (string) "",
    Email: (string) "",
    Company: (string) "",
    Phone: (string) "",
    Line1: (string) "",
    Line2: (string) "",
    Line3: (string) "",
    City: (string) "",
    StateCode: (string) "",
    State: (string) "",
    Country: (string) "",
    Zip: (string) "",
    ValidationStatus: (enums.ValidationStatusEnum) ""
   }
  },
  AmazonPayment: (models.AmazonPayment) {
  },
  Upi: (models.Upi) {
  },
  Paypal: (models.Paypal) {
  },
  Venmo: (models.Venmo) {
  },
  KlarnaPayNow: (models.KlarnaPayNow) {
  },
  Mandates: ([]models.Mandate) <nil>
 },
 (chargebee.PaymentSource) {
  Id: (string) "",
  ResourceVersion: (int64) 0,
  UpdatedAt: (uint64) 0,
  CreatedAt: (uint64) 0,
  CustomerId: (string) "",
  Type: (enums.TypeEnum) "",
  ReferenceId: (string) "",
  Status: (enums.StatusEnum) "",
  Gateway: (enums.GatewayEnum) "",
  GatewayAccountId: (string) "",
  IpAddress: (string) "",
  IssuingCountry: (string) "",
  Deleted: (bool) false,
  BusinessEntityId: (string) "",
  Card: (models.Card) {
  },
  BankAccount: (models.BankAccount) {
  },
  Boleto: (models.CustVoucherSource) {
  },
  BillingAddress: (models.BillingAddress) {
   Address: (models.Address) {
    FirstName: (string) "",
    LastName: (string) "",
    Email: (string) "",
    Company: (string) "",
    Phone: (string) "",
    Line1: (string) "",
    Line2: (string) "",
    Line3: (string) "",
    City: (string) "",
    StateCode: (string) "",
    State: (string) "",
    Country: (string) "",
    Zip: (string) "",
    ValidationStatus: (enums.ValidationStatusEnum) ""
   }
  },
  AmazonPayment: (models.AmazonPayment) {
  },
  Upi: (models.Upi) {
  },
  Paypal: (models.Paypal) {
  },
  Venmo: (models.Venmo) {
  },
  KlarnaPayNow: (models.KlarnaPayNow) {
  },
  Mandates: ([]models.Mandate) <nil>
 }
}
'''
